File: appJava/src/main/java/plby/ChessPiece.java
package plby;

import javax.imageio.ImageIO;
import javax.swing.*;
import java.awt.*;

@SuppressWarnings("unused")
class ChessPiece extends JLabel {
	private final PieceType piece;
	private final int row;
	private final int column;

	ChessPiece(PieceType pieceType, int row, int column) {
		piece = pieceType;
		this.row = row;
		this.column = column;
		try {
			var imagePath = pieceType.getImagePath();
			var imageStream = ChessPiece.class.getResourceAsStream(imagePath);
			var bufferedImage = ImageIO.read(imageStream);
			var scaledImage = bufferedImage.getScaledInstance(58, 58, Image.SCALE_SMOOTH);
			Icon imageIcon = new ImageIcon(scaledImage);
			setIcon(imageIcon);
		} catch (Exception e) {
			e.printStackTrace();
		}
	}

	public int getRow() {
		return row;
	}

	public int getColumn() {
		return column;
	}

	public PieceType getPiece() {
		return piece;
	}
}

File: src/main/kotlin/hummel/ChessPiece.kt
package hummel

import plby.Main
import java.awt.Image
import javax.imageio.ImageIO
import javax.swing.Icon
import javax.swing.ImageIcon
import javax.swing.JLabel

class ChessPiece(val piece: PieceType, val row: Int, val column: Int) : JLabel() {
	init {
		try {
			val imagePath = piece.imagePath
			val imageStream = Main::class.java.getResourceAsStream(imagePath)
			val bufferedImage = ImageIO.read(imageStream)
			val scaledImage = bufferedImage.getScaledInstance(58, 58, Image.SCALE_SMOOTH)
			val imageIcon: Icon = ImageIcon(scaledImage)
			setIcon(imageIcon)
		} catch (e: Exception) {
			e.printStackTrace()
		}
	}
}